### PR TITLE
HW search books

### PR DESCRIPTION
**Pull request by savchenkoarts**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/dto/BookSearchParametersDto.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package mate.academy.onlinebookstore.dto;
+
+public record BookSearchParametersDto(
+        String title,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/dto/BookSearchParametersDto.java:4]: it's better to use array for parameters

File: src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java [hunk 2]
```diff
@@ -25,6 +29,17 @@ public List<BookDto> getAllBooks() {
                 .toList();
     }
 
+    @Override
+    public List<BookDto> searchBooks(BookSearchParametersDto params) {
+        Specification<Book> byTitle =
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java:34]: what if you have 10 search parameters? 
Let's create SpecificationBuilder and SpecificationProvider for each parameter
or you can you BookSpecificationProvider with methods that return particular specificaiton

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/onlinebookstore/specification/SpecificationProvider.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import java.util.List;
+import mate.academy.onlinebookstore.model.Book;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SpecificationProvider {
+    public Specification<Book> getSpecification(List<String> params, String fieldName) {
```
Comment (okuzan) [inline src/main/java/mate/academy/onlinebookstore/specification/SpecificationProvider.java:11]:
```suggestion
    public Specification<Book> getSpecification(List<String> params, String fieldName) {
        if (params == null || params.isEmpty()) {
             return null;
    }

```

**Review by okuzan — APPROVED** [review]

**Pull request merged**